dart/lib/src/sentry_item_type.dart

```diff
@@ -0,0 +1,21 @@
+enum SentryItemType { event, unknown }
```

Comment:
The SDK might be deserializing envelopes written by other SDKs, with types that are not defined in this enum.

Would it be best to have this as a `String` instead and just declare some `const` strings for the known types and fallback to pipe through strings as-is instead?

Reply:
Is this something already in use? The Sentry Java SDK also has a closed enum type to represent this type.

Reply:
as an enum or string, it needs to fall back to a sort of `unknown` type and forward the event to sentry as it is, so both would work for me, you both decide, and yes, Java uses an enum, but its exhaustive as we support every (or almost) item type

Reply:
Waited with adding other cases, as we don't support those items. I'm fine with either way @bruno-garcia

Reply:
If we pick up an envelope from disk, written by another layer or a different version of the SDK, with type `trace-segment` (a new hypothetical type). Is this going to be sent to sentry as `unknown`? If so, that'd be broken.

Reply:
Can't say, as this is not anything that is implemented right now, we don't decode envelopes from file ATM, but i understand your point. The class should provide a way to construct with the correct type string and send it as such, even though it is not part of the enum, correct?

Reply:
Right! Not sure the best way to do this in Dart, but possibly just String. With just a set of const strings for the known types is the way to go